properties.go

```diff
@@ -562,6 +562,31 @@ func (p *Properties) String() string {
 	}
 	return s
 }
+func (p *Properties) FMap(f func(*Properties) interface{}) interface{} {
+	return f(p)
+}
+
+// Transforms properties to a map - useful when users of library are themselves
+// building a library and do not want inner datastructures be exposed to their clients
+// E.g. `Properties` as a return type in library which is using magiconair Properties is not suitable.
+func (p *Properties) ToMap() map[string]string {
+	buffer := make(map[string]string)
+	for k, v := range p.m {
+		buffer[k] = v
+	}
+	return buffer
+}
+
+// Transforms properties to a map - useful when users of library are themselves
+// building a library and do not want inner datastructures be exposed to their clients
+// E.g. `Properties` as a return type in library which is using magiconair Properties is not suitable.
+func (p *Properties) ToMapWith(filters ...func(map[string]string) map[string]string) map[string]string {
```

Comment:
Why not add another `Filter` function to the existing ones instead of creating one that works only with string maps? You can always call `.ToMap()` on the end result. I didn't test the code below but I think you get the drift.

```go
// FilterFunc returns a copy of the properties which includes the values
// which passed all filters.
func (p *Properties) FilterFunc(filters ...func(k,v string) bool) *Properties {
    pp := NewProperties()
outer:
    for k, v := range p.m {
        for _, f := range filters {
            if !f(k, v) {
                continue outer
            }
            pp.Set(k, v)
    }
    return pp
}
```